### PR TITLE
test: increase code coverage

### PR DESCRIPTION
**Pull request by MiahaCybersec**
Related to #40

### CODE REVIEW

**Comment by MiahaCybersec** [comment]
The one failing test is due to mocks relying on `github.com/tonistiigi/fsutil/types`. This is required for the mocks to work. More specifically, this is required for mocking the behaviour of this line:

https://github.com/project-copacetic/copacetic/blob/9da12469013f43505f91afab625b8351b5644fe2/pkg/buildkit/buildkit.go#L88

**Comment by ashnamehrotra** [comment]
@MiahaCybersec could you try running "go mod tidy"?

File: mocks/mock_gwclient.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+	"github.com/opencontainers/go-digest"
+	"github.com/stretchr/testify/mock"
+)
+
+// Mock for gwclient.Client.
+type MockGWClient struct {
+	mock.Mock
+}
+
+func (m *MockGWClient) ResolveSourceMetadata(ctx context.Context, op *pb.SourceOp, opt sourceresolver.Opt) (*sourceresolver.MetaResponse, error) {
```
Comment (ashnamehrotra) [inline mocks/mock_gwclient.go:22]: I don't think we are calling this function?

File: pkg/buildkit/connhelpers/docker_test.go [hunk 2]
```diff
@@ -11,3 +12,41 @@ func TestDocker(t *testing.T) {
...
+		{
+			name:          "Empty addr, DOCKER_HOST env set",
+			addr:          "",
+			envDockerHost: "unix:///var/run/docker.sock",
+			wantErr:       false,
+		},
+		{
+			name:    "Non-empty addr",
+			addr:    "tcp://localhost:2375",
+			wantErr: false,
```
Comment (ashnamehrotra) [inline pkg/buildkit/connhelpers/docker_test.go:32]: could we add a test case where we check for err?

**Review by ashnamehrotra — CHANGES_REQUESTED** [review]

**Comment by MiahaCybersec** [comment]
Had to resolve some conflicts between my branch and main which forced a merge

**Review by ashnamehrotra — APPROVED** [review]
LGTM

**Pull request merged**